.github/workflows/pull-requests.yml

```diff
@@ -590,23 +595,138 @@ jobs:
            fi
           done
 
-      - name: Request clusters info
+      - name: Check if Global L0 node is ready
         run: |
-          sleep 5
-          echo "Dag L1"
-          curl http://127.0.0.1:9100/cluster/info
-          echo ""
-          
-          echo "Currency L0"
-          curl http://127.0.0.1:9400/cluster/info
-          echo ""
-          
-          echo "Currency L1"
-          curl http://127.0.0.1:9700/cluster/info
-          echo ""
+          for ((i = 1; i <= 12; i++)); do
+            if [[ $(curl -s http://127.0.0.1:9000/node/info | jq '.state=="Ready"') == true ]]; then
+              echo "Global L0 is ready"
+              exit 0 
+            else
+              if [ $i -eq 12 ]; then
+                echo "Global L0 is not ready, check the logs"
+                exit 1
+              fi
+              echo "Global L0 not ready yet, trying again in 10s"
+              sleep 10
+            fi
+          done
```

Comment:
there is a lot of repeating code. I'd suggest (either now or in some future PR) to extract the common logic (like execute a code with loop check and exit timeout) and reuse it here and there.